### DISCUSSION TITLE
How to solve this KeyError?

### DISCUSSION THREAD

**Discussion by azure-3** [yasserfarouk/scml-vis#4] [Q&A] [answered]
Hello.

I started using the visualizer, but encountered this KeyError.
I attached all the error message below.

```bash
Exception: 'agent_initial_balances'  
Traceback (most recent call last):
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/scml_vis/presenter.py", line 61, in main
    compiler.main(folder.parent, max_worlds=None)
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/scml_vis/compiler.py", line 799, in main
    ) = get_data(folder)
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/scml_vis/compiler.py", line 703, in get_data
    w, a = get_basic_world_info(base_folder, tname)
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/scml_vis/compiler.py", line 656, in get_basic_world_info
    is_default = dict(zip(winfo["agent_initial_balances"].keys(), itertools.repeat(False)))
KeyError: 'agent_initial_balances'
```

I searched into "info.json", but I couldn’t find "agent_initial_balance".
And searched all of scml files, I could find any "agent_initial_balance".
Could you help me with this?

**Comment by yasserfarouk** [comment, marked as answer]
Hi 
One possible reason for that is if the negmas/scml version used to run the world simulation was not very new. Older versions of scml did not save this key. 

You can get the scml version you have by running:

```shell
scml version
```

Can you please share with me the scml and scml-vis versions you use?
Also, if you could share the log folder itself, I can try to reproduce and resolve the issue.

thanks

**Reply by azure-3** [reply]
My scml version is old(0.4.0), so I updated it and this problem seems to be solved. 
Thank you very much, I really appreciate it.

**Reply by yasserfarouk** [reply]
Great. Nevertheless, I would still like the visualizer to fail gracefully in this case. Will try to reproduce the issue with the same version of scml. thanks
